fuse_graphs/include/fuse_graphs/hash_graph.h

```diff
@@ -330,8 +337,56 @@ class HashGraph : public fuse_core::Graph
    * @param[out] problem The ceres::Problem object to modify
    */
   void createProblem(ceres::Problem& problem) const;
+
+private:
+  // Allow Boost Serialization access to private methods
+  friend class boost::serialization::access;
+
+  /**
+   * @brief The Boost Serialize method that serializes all of the data members in to/out of the archive
+   *
+   * This method, or a combination of save() and load() methods, must be implemented by all derived classes. See
+   * documentation on Boost Serialization for information on how to implement the serialize() method.
+   * https://www.boost.org/doc/libs/1_70_0/libs/serialization/doc/
+   *
+   * @param[in/out] archive - The archive object that holds the serialized class members
+   * @param[in] version - The version of the archive being read/written. Generally unused.
+   */
+  template<class Archive>
+  void serialize(Archive& archive, const unsigned int /* version */)
+  {
+    archive & boost::serialization::base_object<fuse_core::Graph>(*this);
+    archive & constraints_;
+    archive & constraints_by_variable_uuid_;
+    archive & problem_options_;
```

Comment:
I don't think we have to de/serialize the problem options. :thinking:

Reply:
Agreed that it is probably overkill. But if you want to serialize a graph and later load it back and have it behave the same way, then it seems required. And its only a hand full of values, so it's not that painful.

Really what I'm toying with in my head moving a bunch of the graph functionality out into free functions. The graph would build the Problem object (where you could pass in Problem::Options at call time), then optimizing and computing covariances would be generic functions that operate on the Problem. But I have bigger fish to fry at the moment.

Reply:
Fair enough